docs/topics/migrations.txt

```diff
@@ -218,6 +218,38 @@ Apps without migrations must not have relations (``ForeignKey``,
 ``ManyToManyField``, etc.) to apps with migrations. Sometimes it may work, but
 it's not supported.
 
+Swappable dependencies
+======================
+
+A swappable dependency is a feature that allows you to replace or customize
+certain components of the framework with your own implementations. It provides a way
+to define interfaces or abstract classes for specific functionalities, allowing you
+to plug in different implementations based on your requirements.
+
+One of the notable use cases for swappable dependencies is Django's authentication
+system. Django provides a default authentication system, but it also allows you to swap
+it out with a custom authentication backend by defining your own user model and
+authentication backend class. By doing so, you can customize the authentication behavior
+and integrate with external systems or alternative authentication mechanisms.
+
+To make a component swappable, Django relies on the concept of "settings" where you
+can specify the classes or modules to be used for certain functionalities. These settings
+are typically defined in your project's ``settings.py`` file. By modifying
+the relevant settings, you can swap out the default implementations with your own.
+
+Here's an example of how swappable dependencies can be used for authentication::
+
+- Create a custom user model by subclassing ``django.contrib.auth.models.AbstractBaseUser``
+  and ``django.contrib.auth.models.PermissionsMixin``.
+- Define an authentication backend by subclassing ``django.contrib.auth.backends.BaseBackend``
+  and implementing the necessary methods.
+- In your project's settings file, set the ``AUTH_USER_MODEL`` setting to point to your custom
+  user model and specify your custom authentication backend in the ``AUTHENTICATION_BACKENDS`` setting.
+
+Swappable dependencies in Django extend beyond just authentication and can be used in other areas,
+such as the database backend, email backends, cache backends, and more. It provides a powerful
+mechanism for customization and integration within the Django framework.
```

Comment:
There are some issue with rendering this proposition that make it hard to review. Please wrap at 79 chars, fix rendering issues and I will review it again

![image](https://github.com/django/django/assets/2865885/36456cb4-e901-4924-80c2-781258d54e83)

Reply:
Docs also need to be wrapped at 79 chars? 🤔

Reply:
I'd not ask otherwise, see [contributing guide](https://docs.djangoproject.com/en/4.2/internals/contributing/writing-code/coding-style/#python-style):
> _Documentation, comments, and docstrings should be wrapped at 79 characters,..._

Reply:
fixed 👍